### ISSUE TITLE
Graylog System API Timeouts

### ISSUE THREAD

**Issue by amgrice** [Graylog2/graylog2-server#8610]
My Graylog instance is not able to service all requests sent to the /system API endpoint. Some views in the GUI still function i.e. /system/overview but the Node view does not.

## Expected Behavior
I would expect all calls to the /system API endpoint return successfully.

## Current Behavior
Certain calls made to the /system endpoint timeout and fail. Navigating in the UI to /system/nodes/<node-id>/jvm results a 500 error

## Steps to Reproduce (for bugs)

1. Start Graylog 3.3.2.
2. Navigate to the Node's information page OR make an API call to the /system endpoint
3. Observe timeout failure in logs.

## Context

I am attempting to view my Node's information, but it appears that is not currently possible and I am also seeing performance issues as well in the web application. The service is still processing logs and the GUI is somewhat serviceable, but I'm having a hard time finding out exactly what is causing this issue with the system API and what log/verbosity to view/set.

This behavior started a day after upgrading to 3.3.2 from a 3.0.2 install.

I started a conversation in the forums first to try to get traction on this issue, but stalled out: [forum post](https://community.graylog.org/t/graylog-system-api-timeouts/16383)

This seems like a bug somewhere in code, but I don't know enough yet to know where to look. Any help is appreciated.

## Your Environment
Azure instance Standard D8 v3 (8 vcpus, 32 GiB memory) with 12 GB heap size allocated to Graylog. Single instance with TLS on both an Nginx reverse proxy on the host as well as the Graylog application's API.

* Graylog Version: 3.3.2
* Elasticsearch Version: 6.8.6
* MongoDB Version: 4.0.19
* Operating System: CentOS Linux release 7.8.2003 (Core)
* Browser version: Chrome 84.0.4147.89 (Official Build) (64-bit)

**Comment by florianpopp** [comment]
Hi @amgrice could you provide a log file?

**Comment by amgrice** [comment]
Hi @florianpopp I'm pasting in some logs. The AnyExceptionClassMapper error is what gets thrown when going to system/nodes/<node-id> in the web front end:

`2020-07-27T14:31:20.852Z ERROR [AnyExceptionClassMapper] Unhandled exception in REST resource
java.net.SocketTimeoutException: timeout
	at okio.Okio$4.newTimeoutException(Okio.java:232) ~[graylog.jar:?]
	at okio.AsyncTimeout.exit(AsyncTimeout.java:286) ~[graylog.jar:?]
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:241) ~[graylog.jar:?]
	at okio.RealBufferedSource.indexOf(RealBufferedSource.java:358) ~[graylog.jar:?]
	at okio.RealBufferedSource.readUtf8LineStrict(RealBufferedSource.java:230) ~[graylog.jar:?]
	at okhttp3.internal.http1.Http1ExchangeCodec.readHeaderLine(Http1ExchangeCodec.java:242) ~[graylog.jar:?]
	at okhttp3.internal.http1.Http1ExchangeCodec.readResponseHeaders(Http1ExchangeCodec.java:213) ~[graylog.jar:?]
	at okhttp3.internal.connection.Exchange.readResponseHeaders(Exchange.java:115) ~[graylog.jar:?]
	at okhttp3.internal.http.CallServerInterceptor.intercept(CallServerInterceptor.java:94) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142) ~[graylog.jar:?]
	at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:43) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117) ~[graylog.jar:?]
	at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:94) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117) ~[graylog.jar:?]
	at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142) ~[graylog.jar:?]
	at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:88) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117) ~[graylog.jar:?]
	at org.graylog2.rest.RemoteInterfaceProvider.lambda$get$0(RemoteInterfaceProvider.java:61) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:142) ~[graylog.jar:?]
	at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:117) ~[graylog.jar:?]
	at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:229) ~[graylog.jar:?]
	at okhttp3.RealCall.execute(RealCall.java:81) ~[graylog.jar:?]
	at retrofit2.OkHttpCall.execute(OkHttpCall.java:190) ~[graylog.jar:?]
	at org.graylog2.rest.resources.cluster.ClusterSystemResource.jvm(ClusterSystemResource.java:91) ~[graylog.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_252]
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[?:1.8.0_252]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_252]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_252]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory$1.invoke(ResourceMethodInvocationHandlerFactory.java:81) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:144) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:161) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:205) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:99) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:389) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:347) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:102) ~[graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$2.run(ServerRuntime.java:326) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:271) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:267) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:315) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:297) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:267) [graylog.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:317) [graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:305) [graylog.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:1154) [graylog.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:384) [graylog.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:224) [graylog.jar:?]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:181) [graylog.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_252]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_252]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_252]
Caused by: java.net.SocketException: Socket closed
	at java.net.SocketInputStream.read(SocketInputStream.java:204) ~[?:1.8.0_252]
	at java.net.SocketInputStream.read(SocketInputStream.java:141) ~[?:1.8.0_252]
	at sun.security.ssl.InputRecord.readFully(InputRecord.java:465) ~[?:1.8.0_252]
	at sun.security.ssl.InputRecord.read(InputRecord.java:503) ~[?:1.8.0_252]
	at sun.security.ssl.SSLSocketImpl.readRecord(SSLSocketImpl.java:990) ~[?:1.8.0_252]
	at sun.security.ssl.SSLSocketImpl.readDataRecord(SSLSocketImpl.java:948) ~[?:1.8.0_252]
	at sun.security.ssl.AppInputStream.read(AppInputStream.java:105) ~[?:1.8.0_252]
	at okio.Okio$2.read(Okio.java:140) ~[graylog.jar:?]
	at okio.AsyncTimeout$2.read(AsyncTimeout.java:237) ~[graylog.jar:?]
	... 52 more`

The rest of the message are these three continually:

`2020-07-23T14:20:57.303Z WARN  [ProxiedResource] Unable to call https://graylog:9000/api/system/jobs on node <node-id>: timeout
2020-07-23T14:21:09.352Z WARN  [ProxiedResource] Unable to call https://graylog:9000/api/system on node <node-id>: timeout
2020-07-23T14:21:09.354Z WARN  [ProxiedResource] Unable to call https://graylog:9000/api/system/metrics/multiple on node <node-id>: timeout`

**Comment by amgrice** [comment]
If this helps, I've uninstalled 3.3.2 and downgraded to 3.3.1 and 3.3.0, both of which result in the same error.

**Comment by amgrice** [comment]
Anything else I can provide here, @florianpopp ? If there is any more verbose logging I can grab or if there are any dumps needed, happy to provide.

**Comment by amgrice** [comment]
This issue is still present in 3.3.3. Anything else I can do here?

**Comment by amgrice** [comment]
Also, when calling the /api/cluster/<node_id>/jvm endpoint, it _always_ returns a 500 error.

Is there anyway to trace this error?

**Comment by fadenb** [comment]
> Also, when calling the /api/cluster/<node_id>/jvm endpoint, it _always_ returns a 500 error.
> 
> Is there anyway to trace this error?

I am experiencing the same timeouts except for the jvm endpoint. That one works just fine?
```
[<CENSORED-USER>@<CENSORED> ~]$ time http_proxy="" curl -v http://<CENSORED>:9000/api/cluster/bcc504fe-f4f3-4fa6-b6d0-57d12691e5ab/jvm --user <CENSORED-USER>
Enter host password for user '<CENSORED-USER>':
* About to connect() to <CENSORED> port 9000 (#0)
*   Trying 10.147.33.48...
* Connected to <CENSORED> (10.147.33.48) port 9000 (#0)
* Server auth using Basic with user '<CENSORED-USER>'
> GET /api/cluster/bcc504fe-f4f3-4fa6-b6d0-57d12691e5ab/jvm HTTP/1.1
> Authorization: Basic <CENSORED-CREDENTIALS>
> User-Agent: curl/7.29.0
> Host: <CENSORED>:9000
> Accept: */*
>
< HTTP/1.1 200 OK
< X-Graylog-Node-ID: d814f69e-1eee-4577-9e90-ab501f985111
< X-Runtime-Microseconds: 2685203
< Content-Type: application/json
< Date: Thu, 13 Aug 2020 09:08:56 GMT
< Content-Length: 418
<
* Connection #0 to host <CENSORED> left intact
{"free_memory":{"bytes":1120599224,"kilobytes":1094335,"megabytes":1068},"max_memory":{"bytes":2040135680,"kilobytes":1992320,"megabytes":1945},"total_memory":{"bytes":2040135680,"kilobytes":1992320,"megabytes":1945},"used_memory":{"bytes":919580416,"kilobytes":898027,"megabytes":876},"node_id":"bcc504fe-f4f3-4fa6-b6d0-57d12691e5ab","pid":"6471","info":"Oracle Corporation 1.8.0_252 on Linux 3.10.0-1127.el7.x86_64"}
real    0m11.657s
user    0m0.003s
sys     0m0.005s
[<CENSORED-USER>@<CENSORED> ~]$
```

**Comment by amgrice** [comment]
I was able to reproduce this error in Azure using the following steps:

1. Create a new instance using OpenLogic CentOS 7.8|7.7|7.6 image
2. Set SELinux in permissive mode
3. Install MongoDB 4.4 and Graylog 3.3

Each install resulted in the same timeouts.

I then attempted an install on an Ubuntu 18.04-LTS image in Azure. That actually worked and had no timeout issues. Same subnet and resource group.

It would seem that this is an issue with OpenLogic's CentOS image. I have not tried to just do a local install on a vanilla CentOS VM in VirtualBox though.
